src/test/java/com/lithium/dbi/rdbi/recipes/scheduler/DedupJobSchedulerTest.java

```diff
@@ -0,0 +1,318 @@
+        // so it should be ignored.
+        List<TimeJobInfo> reservedJobs = scheduledJobSystem.reserveMulti(tubeName, 1000, 2);
+
+        boolean job2Reserved = false;
+        boolean job4Reserved = false;
+        for(TimeJobInfo jobInfo : reservedJobs) {
+            if(jobInfo.getJobStr().equals("job2")) {
+                job2Reserved = true;
+            } else if(jobInfo.getJobStr().equals("job4")) {
+                job4Reserved = true;
+            } else {
+                fail("Reserved an unexpected job: " + jobInfo.getJobStr());
+            }
+        }
+        assertTrue(job2Reserved);
+        assertTrue(job4Reserved);
+    }
+
+    @Test
+    public void testMultiReservations() {
+        // Schedule a job
+        assertTrue(scheduledJobSystem.schedule(tubeName, "job1", -5000));
+        // Reserve the job so its in the running queue.
+        scheduledJobSystem.reserveSingle(tubeName, 1000);
+
+        // Schedule another job.
+        assertTrue(scheduledJobSystem.schedule(tubeName, "job2", -5000));
+        // Reserve the job so its in the running queue.
+        scheduledJobSystem.reserveSingle(tubeName, 1000);
+
+        assertTrue(scheduledJobSystem.schedule(tubeName, "job1", -10000));  // Should not be able to reserve since its in running queue
+        assertTrue(scheduledJobSystem.schedule(tubeName, "job3", -6000));   // Should be able to reserve first iteration
+        assertTrue(scheduledJobSystem.schedule(tubeName, "job5", -5500));   // Should be able to reserve after job 3 during first iteration
+        assertTrue(scheduledJobSystem.schedule(tubeName, "job2", -5000));   // Should not be able to reserve since its in the running queue
+        assertTrue(scheduledJobSystem.schedule(tubeName, "job6", -2000));   // Should be able to reserve after job 5 during second iteration
+        assertTrue(scheduledJobSystem.schedule(tubeName, "job4", 898789000)); // Not ready to be reserved yet.
+
+        // Reserve 3 jobs. Should return job2, job4 and job7 since they are the next jobs ready to be reserved.
+        // job1 and job2 can't be reserved since they are in the running queue when this reservation request is made
+        List<TimeJobInfo> reservedJobs = scheduledJobSystem.reserveMulti(tubeName, 1000, 3);
+
+        boolean job3Reserved = false;
+        boolean job5Reserved = false;
+        boolean job6Reserved = false;
+        for(TimeJobInfo jobInfo : reservedJobs) {
+            if(jobInfo.getJobStr().equals("job3")) {
+                job3Reserved = true;
+            } else if(jobInfo.getJobStr().equals("job5")) {
+                job5Reserved = true;
+            } else if(jobInfo.getJobStr().equals("job6")) {
+                job6Reserved = true;
+            } else {
+                fail("Reserved an unexpected job: " + jobInfo.getJobStr());
+            }
+        }
+        assertTrue(job3Reserved);
+        assertTrue(job5Reserved);
+        assertTrue(job6Reserved);
```

Comment:
these assertions would be a lot easier if TimeJobInfo implemented .equals or something